src/app.jsx

```diff
@@ -38,11 +38,49 @@ export const Application = () => {
             setCurrentUser(user.name || "");
         })
                 .then(() => {
+                    let result = [];
                     const currentPath = path.join("/");
-                    cockpit.spawn(["ls", "-p", `/home/${currentUser}/${currentPath}`], { superuser: true }).then((res) => {
-                        res = res.split("\n");
-                        setFiles(res.slice(0, res.length - 1));
+                    const channel = cockpit.channel({
+                        payload: "fslist1",
+                        path: `/home/${currentUser}/${currentPath}`,
+                        superuser: "try",
+                        watch: true,
                     });
+
+                    channel.addEventListener("ready", () => {
+                        setFiles(result);
+                        console.log("ready");
+                    });
+
+                    channel.addEventListener("close", (ev, data) => {
+                        console.log("close");
+                    });
```

Comment:
These are no-ops. As long as you don't need them, please don't commit them to production code.